lib/rack/request.rb

```diff
@@ -17,19 +17,15 @@ class << self
     end
 
     self.ip_filter = lambda { |ip| /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i.match?(ip) }
-
-    # The environment of the request.
-    attr_reader :env
-
     ALLOWED_SCHEMES = %w(https http wss ws).freeze
     SCHEME_WHITELIST = ALLOWED_SCHEMES
     if Object.respond_to?(:deprecate_constant)
       deprecate_constant :SCHEME_WHITELIST
     end
 
     def initialize(env)
-      @env = env
       @params = nil
+      super(env)
```

Comment:
I think it would be fine to remove the `super(env)` and just leave it at `@env = env`.  The `super` call in `Env#initialize` is needed for Rails, but we don't need to call into `Env#initialize` here.